cmake/templates/oneDPLConfig.cmake.in

```diff
@@ -25,8 +25,10 @@ if (EXISTS "${_onedpl_headers}")
 
         # This compiler check/requirement does not work on Windows with dpcpp and dpcpp-cl compiler drivers
         # icx-cl and icpx are not affected
-        if (NOT (WIN32 AND CMAKE_CXX_COMPILER MATCHES ".*dpcpp(-cl)?(.exe)?$"))
+        if (NOT (WIN32 AND CMAKE_CXX_COMPILER_ID MATCHES "IntelLLVM Workaround"))
             target_compile_features(oneDPL INTERFACE cxx_std_17)
+        else()
+            target_compile_options(oneDPL INTERFACE -std=c++17)
```

Comment:
This is a good point.  This needs some work.

Reply:
It seems we can tell cmake that the feature is supported even by a compiler with a custom id and thus avoid this branch.

I performed some experiment with inclusion of `cmake/windows-dpcpp-toolchain.cmake` with this additional line:
```bash
set(CMAKE_CXX17_STANDARD_COMPILE_OPTION "-std=c++17")
```
`target_compile_features(oneDPL INTERFACE cxx_std_17)` worked after that. 

Currently, I do not exactly know how it works, but I faced these definition in toolchains from cmake repository, e.g.: https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Compiler/MSVC-CXX.cmake. I am investigating it.

Reply:
Sadly, with this extra line, I am still seeing the same error:
```
CMake Error in test/CMakeLists.txt:
  No known features for CXX compiler

  "IntelLLVM GNU-Like Workaround"

  version 2023.0.0.
```

I'm trying to dig into cmake to understand what else we have to set to get this to work.  It would be ideal to not have to special case this check later on.